File: PHP_POO/Aula 04/Caneta.php
<?php

class Caneta {
    public $modelo;
    private $cor;
    private $ponta;
    protected $carga;
    protected $tampada;


    //Metodo Construtor
    /*public function __construct(){
        $this->cor = "Azul";
        $this->tampar();
    }*/
    //O próximo exemplo é o metodo construtor com passagem de parâmetros.
    public function __construct($m, $c, $p){
        $this->modelo = $m;
        //$this->setModelo($m);  Pode ser utilizado também com setter.
        $this->cor = $c;
        $this->ponta = $p;
        $this->tampar();
    }

    public function tampar(){
        $this->tampada = true;
    }

     //Metodos Getters e Setters
    public function getModelo(){
        return $this->modelo;
    }
    public function setModelo($m){
        $this->modelo = $m;
    }
    public function getPonta(){
        return $this->ponta;
    }
    public function setPonta($p){
        $this->ponta = $p;
    }


}

?>

File: JAVA_POO/aula04/Caneta.java
package aula04;

public class Caneta {
    private String modelo;
    private float ponta;
    private String cor;
    private boolean tampada;

    //-------------------------------------------- METODOS CONSTRUTOR -------------------------------------------//

    public Caneta(String m, String c, float p) {
        this.modelo = m;
        this.cor = c;
        this.ponta = p;
        this.tampar();
    }


    //---------------------------------------- METODOS GETTERS AND SETTERS --------------------------------------//


    public String getModelo() {
        return modelo;
    }

    public void setModelo(String m) {
        this.modelo = m;
    }

    public float getPonta() {
        return ponta;
    }

    public void setPonta(float p) {
        this.ponta = p;
    }

    public boolean isTampada() {
        return tampada;
    }

    public void setTampada(boolean tampada) {
        this.tampada = tampada;
    }

    public String getCor() {
        return cor;
    }

    public void setCor(String cor) {
        this.cor = cor;
    }


    //----------------------------------------------- MÉTODOS ---------------------------------------------------//


    public void status() {
        System.out.println("Modelo: " + this.getModelo());
        System.out.println("Ponta: " + this.getPonta());
        System.out.println("Cor: " + this.getCor());
        System.out.println("A caneta esta tampada? " + this.isTampada());
    }

    public void tampar(){
        this.tampada = true;
    }

    public void destampar(){
        this.tampada = false;
    }

}